### ISSUE TITLE
Export Integration to e.g. Labelstudio

### ISSUE THREAD

**Issue by JWittmeyer** [code-kern-ai/refinery#143]
**Is your feature request related to a problem? Please describe.**
I want to use the data from labelstudio in refinery or the other way around

**Describe the solution you'd like**
when exporting records (not the project snapshot). I want to be able to select an export format (current, for labelstudio, ....), maybe what to export (e.g. only data, specific tasks etc.) and what file type (.csv, .json ,....)

**Describe alternatives you've considered**
crating a parser on my own

**Additional context**
\-

**Comment by JWittmeyer** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Kern-Dev-62e7b2259871ff54182d813c/issues/code-kern-ai/refinery/143?planning-poker) with Zenhub @JWittmeyer @SirDegraf

**Comment by JWittmeyer** [comment]
##Idea proposal

### Preparations (imo can be done in parallel):
- [ ] Analyze format Label Studio and prepare Mapping Document (6 h)
- [ ] Think of UI Design (2 h)
- [ ] Think of Backend Design (4 h)
- [ ] Create Issues/PRs for Repos (best guess: UI, Gateway, Submodule model, Proxy & SDK) (1h)

### Findings Analyzation (classification)
1. Only one "labeling task" per project in Label Studio?
2. Task Information (settings) isn't part of the label studio export but can be prepared for copy paste
3. Task Values are only correctly transferred/assigned to the task in label studio if the .json format is chosen, csv treats the label column as normal text column
4. Import can be prase from csv to kern.ai import only if we can "select" what columns are the label values
5. Import of .json unproblematic, however slight loss of values (e.g. modified at)
6. annotators need to be mapped (export only holds an id)



### Design UI
Export records should be able to select what we want to export and how.
What: 
- (Rows) static slice, dynamic slice or current data browser selection
- (Columns) key (UUID or row number), "attribute data", N labeling tasks

How:
- File Type (.json, .csv, .excel sheet)
- Format (current, Labelstudio,...)
   - E.g. Label studio only allows one labeling task selected (or multiple files)

Import
- Select between import templates (current, Label Studio csv, Label Studio json, ...)
- Decide on update behavior (can be a different issue/sprint)
    - (atm all columns need to be present to add new columns)
    - "overwrite" mode -> all values present in the given document are used to update existing values
    - "append" mode -> only values != "" || None are used to replace values / appended
- Depending on the template different options are possible
    - current -> file import option ala pandas 
    - Label Studio csv 
        -> "Columns that should be treated as label"
        -> Map label studio user-id to kern user-id
    - Label Studio json
        -> Map label studio user-id to kern user-id
- Maybe store presets of import configurations so one doesn't need to reapply them every single time?
- Maybe we should also add some logic to analyze the selected file before processing it
    - csv first 100 lines
    - json if < 5 MB completely and remove all but 100 entries?
    - or sent the file to the backend and analyze it there? (probably best option)


### Design Backend
<placeholder>

**Comment by JWittmeyer** [comment]
Approach:
- first we only do the export from kern (including export options from kern to import into label studio)
- Depending on the time passed we start with the import this or next sprint
- UI @JWittmeyer 
- Backend @SirDegraf 
- SDK & Proxy (after UI and Backend)

Steps:
- [ ] Analyze extraction label studio tasks
- [ ] Interface decision (between UI & Gateway)
- [ ] concept UI
- [ ] concept Gateway
- [ ] implement
- [ ] QA

**Comment by JWittmeyer** [comment]
Export changes are included in PR https://github.com/code-kern-ai/refinery-gateway/pull/69.
New issue will be created for import & other integrations than label sutdio
